lib/_http_agent.js

```diff
@@ -319,6 +326,22 @@ function installListeners(agent, s, options) {
   }
   s.on('close', onClose);
 
+  function onTimeout() {
+    debug('CLIENT socket onTimeout');
+
+    const sockets = agent.freeSockets;
+    for (const name of ObjectKeys(sockets)) {
+      const index = sockets[name].indexOf(s);
+      if (index !== -1) {
+        sockets[name].splice(index, 1);
+        if (sockets[name].length === 0)
+          delete sockets[name];
```

Comment:
@lundibundi We still need `name` here... I prefer avoiding `for in`. I think V8 is pretty good at avoiding the unnecessary array here, it's usually `Object.entries` that cause perf pitfalls.

Reply:
Oh yeah, I meant `entries`.
I don't think it will be possible to avoid creating an array for keys since we modify the object in the loop.
Anyway, just a suggestion.